Modules/Core/Common/include/itkByteSwapper.hxx

```diff
@@ -39,202 +39,170 @@ namespace itk
 //------Big Endian methods----------------------------------------------
 
 // Use different swap methods based on type
-#ifdef CMAKE_WORDS_BIGENDIAN
-template <typename T>
-void
-ByteSwapper<T>::SwapFromSystemToBigEndian(T *)
-{}
-#else
+
 template <typename T>
 void
 ByteSwapper<T>::SwapFromSystemToBigEndian(T * p)
 {
-  switch (sizeof(T))
+  if constexpr (!m_SystemIsBigEndian)
   {
-    case 1:
-      return;
-    case 2:
-      Self::Swap2(p);
-      return;
-    case 4:
-      Self::Swap4(p);
-      return;
-    case 8:
-      Self::Swap8(p);
-      return;
-    default:
-      itkGenericExceptionMacro("Cannot swap number of bytes requested");
+    switch (sizeof(T))
+    {
+      case 1:
+        return;
+      case 2:
+        Self::Swap2(p);
+        return;
+      case 4:
+        Self::Swap4(p);
+        return;
+      case 8:
+        Self::Swap8(p);
+        return;
+      default:
+        itkGenericExceptionMacro("Cannot swap number of bytes requested");
+    }
   }
 }
 
-#endif
-
-#ifdef CMAKE_WORDS_BIGENDIAN
-template <typename T>
-void
-ByteSwapper<T>::SwapRangeFromSystemToBigEndian(T *, BufferSizeType)
-{
-  // nothing needs to be done here...
-}
-
-#else
-
-#  ifdef __INTEL_COMPILER
-#    pragma warning disable 280 // remark #280: selector expression is constant
-#  endif
 template <typename T>
 void
 ByteSwapper<T>::SwapRangeFromSystemToBigEndian(T * p, BufferSizeType num)
 {
-  switch (sizeof(T))
+  if constexpr (!m_SystemIsBigEndian)
   {
-    case 1:
-      return;
-    case 2:
-      Self::Swap2Range(p, num);
-      return;
-    case 4:
-      Self::Swap4Range(p, num);
-      return;
-    case 8:
-      Self::Swap8Range(p, num);
-      return;
-    default:
-      itkGenericExceptionMacro("Cannot swap number of bytes requested");
+    switch (sizeof(T))
```

Comment:
I was initially going to comment that this switch statement is duplicated code and could be factored into a function... But isn't sizeof(T) known at compile time? We should be able to do a better Swap range function to avoid duplication.

Reply:
@blowekamp It looks like the existing `Swap2`, `Swap4`, and `Swap8` could be combined into some "SwapN" member function, like:

```cpp
void
ByteSwapper<T>::SwapN(void * ptr)
{
  auto * p = static_cast<char *>(ptr);

  for (size_t i{}; i < (N / 2); ++i)
  {
    std::swap(p[i], p[(N - 1) - i]);
  }
}
```
Would you like that? Anyway, maybe nice for a follow-up pull request  😃 

For comparison, the old `Swap8` implementation is at: https://github.com/InsightSoftwareConsortium/ITK/blob/e9b3d24f782a42f5586169e048b8d289f869d78a/Modules/Core/Common/include/itkByteSwapper.hxx#L350-L361